### PR TITLE
BOOKSTORE: Add reducers and actions

### PR DESCRIPTION
**Pull request by ndorrh**
### BookStore

In this milestone, I did the following:

- Install the Redux toolkit using npm
- Add a redux store
- Add a reducer for books
- Add a reducer for categories
- Import store in  index.js and wrap App.js in Provider and gave it a prop of rootReducer
- Use professional git flow

### CODE REVIEW

File: src/components/InputBook.js [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
 function InputBook() {
+  const dispatch = useDispatch();
+  const handleAddBook = () => {
+    dispatch(booksActions.addBook({
+      id: 1,
+      title: 'Awa and Other plays',
+      author: 'Ndorrh Oswald',
+
+    }));
+  };
```
Comment (m05e5) [inline src/components/InputBook.js]: Hi there. Normally you're supposed to dispatch an object with the data from the input fields and a new ID. Currently, you just add some predefined data with a constant id. So when we remove one the added books, all of them are deleted
 - [ ] Dispatch data from the input fields
 - [ ] Not two books are supposed to have the same ID so please kindly fix it

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @ndorrh , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Required Changes ♻️
* You really did a great job but there are some features you need to improve

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ndorrh** [comment]
@m05e5
Thanks a lot for the review.
I will implement the optional changes

I didn't implement that because it is what we are going to be working on today

File: src/components/InputBook.js [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
 
 function InputBook() {
+  const dispatch = useDispatch();
+  const handleAddBook = () => {
+    dispatch(booksActions.addBook({
+      id: 1,
+      title: 'Awa and Other plays',
+      author: 'Ndorrh Oswald',
+
+    }));
```
Comment (brytebee) [inline src/components/InputBook.js]: - [ ] **Required:** As stated by the previous reviewer, I have added a part of the requirement, you need to implement it such that a new state is returned when the `addBook` action is fired.

image

Currently, what is added in the form isn't return but the hardcoded data, Kindly fix this.

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @ndorrh👋, 

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️

##

Great work on making the changes 👏 👏 
You've done well implementing some of the requested changes requested by the previous review, but there are still some which aren't addressed yet.


# Suggested changes
_Check the comments under the review._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/components/InputBook.js (hunk 2 above)
Comment (m05e5) [inline src/components/InputBook.js]: ## 💡💡💡This is for the next reviewer💡💡💡
- [ ] Hi it's like there is a misunderstanding. I just realized that for this milestone the student was just supposed to do the reducers and actions so, This part is for the next milestone and can be ignored for this PR. Sorry for the misunderstanding.

Comment (ndorrh) [reply src/components/InputBook.js]: @brytebee  thanks for the review but I think this is not part of yesterdays project.
Please check the requirements of the project of yesterday.
I summited yesterday's projet and not today. 
Please kindly make the review base on the project requirements of yesterday

**Review by t-yanick — APPROVED** [review]
Hi @ndorrh, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**